databricks/koalas/frame.py

```diff
@@ -7521,6 +7521,146 @@ def keys(self):
         """
         return self.columns
 
+    # TODO: axis = 1
+    def idxmax(self, axis=0):
+        """
+        Return index of first occurrence of maximum over requested axis.
+        NA/null values are excluded.
+
+        Parameters
+        ----------
+        axis : 0 or 'index'
+            Can only be set to 0 at the moment.
+
+        Returns
+        -------
+        Series
+
+        See Also
+        --------
+        Series.idxmax
+
+        Examples
+        --------
+        >>> kdf = ks.DataFrame({'a': [1, 2, 3, 2],
+        ...                     'b': [4.0, 2.0, 3.0, 1.0],
+        ...                     'c': [300, 200, 400, 200]})
+        >>> kdf
+           a    b    c
+        0  1  4.0  300
+        1  2  2.0  200
+        2  3  3.0  400
+        3  2  1.0  200
+
+        >>> kdf.idxmax()
+        a    2
+        b    0
+        c    2
+        Name: 0, dtype: int64
+
+        For Multi-column Index
+
+        >>> kdf = ks.DataFrame({'a': [1, 2, 3, 2],
+        ...                     'b': [4.0, 2.0, 3.0, 1.0],
+        ...                     'c': [300, 200, 400, 200]})
+        >>> kdf.columns = pd.MultiIndex.from_tuples([('a', 'x'), ('b', 'y'), ('c', 'z')])
+        >>> kdf
+           a    b    c
+           x    y    z
+        0  1  4.0  300
+        1  2  2.0  200
+        2  3  3.0  400
+        3  2  1.0  200
+
+        >>> kdf.idxmax()
+        (a, x)    2
+        (b, y)    0
+        (c, z)    2
+        Name: 0, dtype: int64
+        """
+        from databricks.koalas.series import Series
+        sdf = self._sdf
+        max_cols = map(lambda x: F.max(x).alias(x), self._internal.data_columns)
+        sdf_max = sdf.select(*max_cols)
+
+        col_idxmax_dict = dict()
+        for column_name, max_val in zip(sdf_max.columns, sdf_max.head()):
```

Comment:
We should deduplicate the logic with `Series.idx` such as https://github.com/databricks/koalas/pull/678 (although I don't think we need a separate private method like `_idx`).

Can we document that this will requires to sort for each column in a `... note ::`?